lib/krikri/util/extended_date_parser.rb

```diff
@@ -21,17 +21,18 @@ module ExtendedDateParser
     #
     # @return [Date, EDTF::Epoch, EDTF::Interval, nil] the date parsed or nil
     def parse(date_str, allow_interval = false)
-      date_str.strip!
-      date_str.gsub!(/\s+/, ' ')
-      date = parse_interval(date_str) if allow_interval
-      date ||= parse_m_d_y(date_str)
-      date ||= Date.edtf(date_str.gsub('.', '-'))
-      date ||= partial_edtf(date_str)
-      date ||= decade_hyphen(date_str)
-      date ||= month_year(date_str)
-      date ||= decade_s(date_str)
-      date ||= hyphenated_partial_range(date_str)
-      date ||= parse_date(date_str)
+      str = preprocess(date_str.dup)
+      date = parse_interval(str) if allow_interval
+      date ||= parse_m_d_y(str)
+      date ||= Date.edtf(str.gsub('.', '-'))
+      date ||= partial_edtf(str)
+      date ||= decade_hyphen(str)
+      date ||= month_year(str)
+      date ||= decade_s(str)
+      date ||= hyphenated_partial_range(str)
+      date ||= parse_date(str)
+      date ||= circa(str) if str.match(/[circabout]/i)
```

Comment:
Right, `/[circabout]/i` matches any one of those characters in the string, such that it would match most month names  or abbreviations.